src/SSHDebugPS/CommandRunner.cs

```
@@ -9,304 +9,304 @@
+
+            if (_processStartInfo == null)
             {
-                ErrorOccured?.Invoke(this, new ErrorOccuredEventArgs(e));
-                Dispose();
+                throw new InvalidOperationException("Unable to create process. Process start info does not exist");
             }
-        }
 
-        private void OnErrorOutput(object sender, DataReceivedEventArgs e)
-        {
-            if (!string.IsNullOrWhiteSpace(e.Data))
+            lock (_lock)
             {
-                ErrorOccured?.Invoke(this, new ErrorOccuredEventArgs(new Exception(e.Data)));
+                _process = new System.Diagnostics.Process();
+                _process.StartInfo = _processStartInfo;
+
+                _process.Exited += OnProcessExited;
+                _process.EnableRaisingEvents = true;
+
+                _process.Start();
             }
+
+            _stdInWriter = new StreamWriter(_process.StandardInput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), BUFMAX);
+            _stdOutReader = new StreamReader(_process.StandardOutput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), false, BUFMAX);
+            _stdErrReader = new StreamReader(_process.StandardError.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), false, BUFMAX);
+
+            _outputReadLoopTask = System.Threading.Tasks.Task.Run(() => ReadLoop(_stdOutReader, _cancellationSource.Token, true, OnOutputReceived));
+            _errorReadLoopTask = System.Threading.Tasks.Task.Run(() => ReadLoop(_stdErrReader, _cancellationSource.Token, false, OnErrorReceived));
         }
 
-        private void OnProcessOutput(object sender, DataReceivedEventArgs e)
+        public void Write(string text)
         {
             lock (_lock)
             {
-                OutputReceived?.Invoke(this, e.Data);
-                if (_hasExited && _localProcess.StandardOutput.EndOfStream)
+                if (IsRunning())
```

>if (IsRunning()) [](start = 16, length = 16)

Is there a reason this silently drops input if the command isn't running, but the SSH version throws an exception?  Seems like the behavior should be consistent.